src/Integration.Vsix/SonarLintTracker/TaggerProvider.cs

```diff
@@ -0,0 +1,240 @@
+    ///
+    /// It is also the <see cref="ITableDataSource"/> that reports Sonar errors.
+    /// </summary>
+    //TODO Temporarily disabled. Uncomment the next lines when the daemon installer is in place
+    //[Export(typeof(IViewTaggerProvider))]
+    [TagType(typeof(IErrorTag))]
+    [ContentType("text")]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    internal sealed class TaggerProvider : IViewTaggerProvider, ITableDataSource
+    {
+        internal readonly ITableManager ErrorTableManager;
+        internal readonly ITextDocumentFactoryService TextDocumentFactoryService;
+
+        private readonly List<SinkManager> managers = new List<SinkManager>();
+        private readonly TrackerManager trackers = new TrackerManager();
+
+        internal static TaggerProvider Instance { get; private set; }
+
+        [ImportingConstructor]
+        internal TaggerProvider([Import] ITableManagerProvider provider, [Import] ITextDocumentFactoryService textDocumentFactoryService)
+        {
+            this.ErrorTableManager = provider.GetTableManager(StandardTables.ErrorsTable);
+            this.TextDocumentFactoryService = textDocumentFactoryService;
+
+            this.ErrorTableManager.AddSource(this, StandardTableColumnDefinitions.DetailsExpander,
+                                                   StandardTableColumnDefinitions.ErrorSeverity, StandardTableColumnDefinitions.ErrorCode,
+                                                   StandardTableColumnDefinitions.ErrorSource, StandardTableColumnDefinitions.BuildTool,
+                                                   StandardTableColumnDefinitions.ErrorSource, StandardTableColumnDefinitions.ErrorCategory,
+                                                   StandardTableColumnDefinitions.Text, StandardTableColumnDefinitions.DocumentName,
+                                                   StandardTableColumnDefinitions.Line, StandardTableColumnDefinitions.Column,
+                                                   StandardTableColumnDefinitions.ProjectName);
+
+            TaggerProvider.Instance = this;
+        }
+
+        /// <summary>
+        /// Create a tagger that will track Sonar issues on the view/buffer combination.
+        /// </summary>
+        public ITagger<T> CreateTagger<T>(ITextView textView, ITextBuffer buffer) where T : ITag
+        {
+            // Only attempt to track the view's edit buffer.
+            // Multiple views could have that buffer open simultaneously, so only create one instance of the tracker.
+            if ((buffer == textView.TextBuffer) && (typeof(T) == typeof(IErrorTag)))
+            {
+                ITextDocument document;
+                if (TextDocumentFactoryService.TryGetTextDocument(textView.TextDataModel.DocumentBuffer, out document))
+                {
+                    var path = document.FilePath;
+                    // TODO find a better way to detect JavaScript
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)